middleware/common/include/common/communication/ipc/send.h

```diff
@@ -157,12 +158,67 @@ namespace casual
 
          private:
 
-            ipc::outbound::partial::Destination m_destination;
+            ipc::partial::Destination m_destination;
             queue_type m_queue;
          };
 
          strong::correlation::id send( const strong::ipc::id& ipc, Message&& message);
 
+         //! Take care of devices that can "reconnect"
+         template< typename D>
+         auto send_dispatch( D& device, Message&& message, traits::priority::tag< 1>) 
+            -> decltype( void( device.connector().reconnect()), strong::correlation::id{})
+         {
+            //! TODO: this does not feel that good, at all.. The whole "reconnect" thing should
+            //!   be at "one place". Now we we have this copy... Try to consolidate...
+
+            auto result = message.complete.correlation();
+
+            if( auto found = algorithm::find( m_destinations, device.connector().process().ipc))
+            {
+               if( found->push( std::move( message)).send( *m_directive))
+                  m_destinations.erase( std::begin( found));
+
+               return result;
+            }
+            else
+            {
+               while( true)
```

Comment:
There is not a risk for a never ending loop here?

Reply:
It's to keep going if we've get multiple interups... (the reconnect throws if fail) The whole exception paradigm does not work well when the "errors" are expected... I'm going to change at least the whole _device_ API to communicate with _std::expected_ instead. Hast to be a lot better, since a lot of the "errors" are in dead expected...